docs/codeql/codeql-language-guides/customizing-library-models-for-javascript.rst

```diff
@@ -0,0 +1,474 @@
+.. _customizing-library-models-for-javascript:
+
+:orphan:
+:nosearch:
+
+Customizing Library Models for JavaScript
+=========================================
+
+.. include:: ../reusables/beta-note-customizing-library-models.rst
+
+The JavaScript analysis can be customized by adding library models in data extension files.
+
+A data extension for JavaScript is a YAML file of form:
+
+.. code-block:: yaml
+
+  extensions:
+    - addsTo:
+        pack: codeql/javascript-all
+        extensible: <name of extension point>
+      data:
+        - <tuple1>
+        - <tuple2>
+        - ...
+
+The data extension can contribute to the following extension points:
```

Comment:
The [Java PR](https://github.com/github/codeql/pull/12410/files) used this language:

> "The CodeQL library for Java exposes the following extensible predicates:"

Could we have this here?

```suggestion
The CodeQL library for JavaScript exposes the following extensible predicates:
```